tests/processor/test_main.py

```diff
@@ -291,6 +291,34 @@ def test_create_or_update_source_location():
     assert sl.id == sl2.id
 
 
+@mock.patch('assayist.processor.base.Analyzer.conditional_connect')
+def test_create_or_update_source_location_bad(m_conditional_connect):
+    """Test that create_or_update_source_location rolls back on error."""
+    m_conditional_connect.side_effect = ValueError('something broke')
+
+    rpm_comp = Component.get_or_create({
+        'canonical_namespace': 'a',
+        'canonical_name': 'test',
+        'canonical_type': 'rpm'})[0]
+    rpm_comp.save()
+    analyzer = main_analyzer.MainAnalyzer()
+    url = 'www.whatever.com'
+    canonical_version = '0-1-3'
+
+    raised = False
+    try:
```

Comment:
This works, but you can use this to test exceptions instead:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions